tests/test_core.py

```diff
@@ -171,6 +170,136 @@ def test_mass(Q, T):
     npt.assert_almost_equal(left, right)
 
 
+@pytest.mark.parametrize("Q, T", test_data)
+def test_mass_nan(Q, T):
+    T[1] = np.nan
+    m = Q.shape[0]
+
+    left = np.linalg.norm(
+        core.z_norm(core.rolling_window(T, m), 1) - core.z_norm(Q), axis=1
+    )
+    left[np.isnan(left)] = np.inf
+
+    right = core.mass(Q, T)
+    npt.assert_almost_equal(left, right)
+
+
+@pytest.mark.parametrize("Q, T", test_data)
+def test_mass_inf(Q, T):
+    T[1] = np.inf
+    m = Q.shape[0]
+
+    left = np.linalg.norm(
+        core.z_norm(core.rolling_window(T, m), 1) - core.z_norm(Q), axis=1
+    )
+    left[np.isnan(left)] = np.inf
+
+    right = core.mass(Q, T)
+    npt.assert_almost_equal(left, right)
+
+
+def test_apply_exclusion_zone():
+    T = np.array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], dtype=float)
+    exclusion_zone = 2
+
+    index = 1
+    left = np.array([np.inf, np.inf, np.inf, np.inf, 4, 5, 6, 7, 8, 9])
+    right = T.copy()
+    core.apply_exclusion_zone(right, index, exclusion_zone)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_array_equal(left, right)
+
+    index = 8
+    left = np.array([0, 1, 2, 3, 4, 5, np.inf, np.inf, np.inf, np.inf])
+    right = T.copy()
+    core.apply_exclusion_zone(right, index, exclusion_zone)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_array_equal(left, right)
+
+    index = 4
+    left = np.array([0, 1, np.inf, np.inf, np.inf, np.inf, np.inf, 7, 8, 9])
+    right = T.copy()
+    core.apply_exclusion_zone(right, index, exclusion_zone)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_array_equal(left, right)
+
+
+def test_apply_exclusion_zone_multidimensional():
```

Comment:
Same as above: I think it is probably important to test the "edge cases: where index = 0 and index = 9